syntax = "proto3";

//import "login_id.proto";

package LoginProtocol;

message GameInfo {
    int32 gameid = 1;
    int32 platform = 2;
    bytes name = 3;
    bytes appid = 4;
    bytes secret = 5;
}

message C2S_LoginLogin {
    bytes loginname = 1;
}

message S2C_LoginLogin {
    int32 ret = 1;
    int64 role_id = 2;
}

message C2S_LoginCreate {
    bytes loginname = 1;
    bytes role_name = 2;
    bytes avatarurl = 3;
}

message S2C_LoginCreate {
    int32 ret = 1;
    int64 role_id = 2;
}

message C2S_LoginLoad {
    int64 role_id = 1;
}

message S2C_LoginLoad {
    int32 ret = 1;
}

message C2S_LoginOpenid {
    int32 gameid = 1;
    bytes code = 2;
}

message S2C_LoginOpenid {
    int32 ret = 1;
    bytes openid = 2;
}